paymentsheet/src/main/java/com/stripe/android/customersheet/util/CustomerSheetHacks.kt

```diff
@@ -28,23 +28,32 @@ internal object CustomerSheetHacks {
     val adapter: Deferred<CustomerAdapter>
         get() = _adapter.asDeferred()
 
-    private val _dataSource = MutableStateFlow<CombinedDataSource<*>?>(null)
-
+    private val _paymentMethodDataSource = MutableStateFlow<CustomerSheetPaymentMethodDataSource?>(null)
     val paymentMethodDataSource: Deferred<CustomerSheetPaymentMethodDataSource>
-        get() = _dataSource.asDeferred()
+        get() = _paymentMethodDataSource.asDeferred()
 
+    private val _savedSelectionDataSource = MutableStateFlow<CustomerSheetSavedSelectionDataSource?>(null)
     val savedSelectionDataSource: Deferred<CustomerSheetSavedSelectionDataSource>
-        get() = _dataSource.asDeferred()
+        get() = _savedSelectionDataSource.asDeferred()
 
+    private val _intentDataSource = MutableStateFlow<CustomerSheetIntentDataSource?>(null)
     val intentDataSource: Deferred<CustomerSheetIntentDataSource>
-        get() = _dataSource.asDeferred()
+        get() = _intentDataSource.asDeferred()
```

Comment:
I don't necessarily like that this class has become a dependency manager. We definitely need to find a better solution for this. I'm going to look into this post-Customer Session private beta and just make sure that this is well tested.